cli/internal/providers/retl/sqlmodel/model.go

```diff
@@ -0,0 +1,73 @@
+package sqlmodel
+
+import "fmt"
+
+// ResourceType is the type identifier for SQL Model resources
+const (
+	ResourceType    = "retl-source-sql-model"
+	ModelSourceType = "model"
+
+	LocalIDKey              = "local_id"
+	DisplayNameKey          = "display_name"
+	DescriptionKey          = "description"
+	AccountIDKey            = "account_id"
+	PrimaryKeyKey           = "primary_key"
+	SourceDefinitionNameKey = "source_definition_name"
+	EnabledKey              = "enabled"
+	SQLKey                  = "sql"
+	SourceIDKey             = "source_id"
+	SourceTypeKey           = "source_type"
+	CreatedAtKey            = "created_at"
+	UpdatedAtKey            = "updated_at"
+)
+
+// SQLModelSpec represents the YAML specification for a SQL Model resource
+type SQLModelSpec struct {
+	ID                   string  `mapstructure:"id"`
+	DisplayName          string  `mapstructure:"display_name"`
+	Description          string  `mapstructure:"description"`
+	File                 *string `mapstructure:"file"`
+	SQL                  *string `mapstructure:"sql"`
+	AccountID            string  `mapstructure:"account_id"`
+	PrimaryKey           string  `mapstructure:"primary_key"`
+	SourceDefinitionName string  `mapstructure:"source_definition_name"`
```

Comment:
That's an interesting point we need to discuss: The original PRD did not assume the user including a "source_definition_name" in the YAML. I'm guess you added this because we need to know the source definition name by the time we make the API call (although this is not an updatable field).

We need to figure out:
- is "source_definition_name" the best name for the YAML context?
- how do we prevent users from trying to update it, e.g it should result in some error instead of reporting a potential diff.